daffodil-io/src/main/scala/org/apache/daffodil/processors/charset/BitsCharsetDefinition.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.daffodil.processors.charset
+
+/**
+ * Must be implemented by all charsets.
+ *
+ * These are the classes which must be dynamically loaded in order to add a charset implementation
+ * to Daffodil. All charsets must implement this class
+ *
+ * A saved processor does NOT serialize this class. It calls the newFactory method and serializes
+ * the resulting BitsCharsetFactory.
+ */
+abstract class BitsCharsetDefinition {
+
+  def name(): String
+
+  def newFactory(): BitsCharsetFactory
```

Comment:
Is there a use case for the Factory indirection? Based on the comment it seems to be related to serialization, but I don't see anywhere where we store the created factories where they could be serialized.

The only use I see is `find(...).newFactory().newInstance()`. So we create a factory, create an instance from that factory, and throw away the factory. I wonder if we can simplify this and just treat `BitsCharsetDefinition` as if it were the factory--it can have the `newInstance` function that is directly called when a new instance of a charset is needed?

Reply:
Well, the comments about serialization were copied from the LayerCompiler classes.  It probably is true that Daffodil processors store / serialize layer classes in such a way that this indirection makes sense, but it may not be true that Daffodil processors need this indirection for BitCharset classes at all.  @alexanderrevello, try removing the indirection and see if all the tests (including integration/CLI tests) still pass.  If so, then Steve is right and the indirection is unneeded.

Reply:
Indirection is not needed and has been removed